OpenSim/Moco/MocoGoal/MocoContactImpulseTrackingGoal.cpp

```diff
@@ -86,24 +85,19 @@ void MocoContactTrackingGoal::addScaleFactor(const std::string& name,
         }
     }
     OPENSIM_THROW_IF_FRMOBJ(!foundContactGroup, Exception,
-            "Contact group associated with external force '{}' not found.",
-            externalForceName);
-
-    // Check that the index is in the correct range.
-    OPENSIM_THROW_IF_FRMOBJ((index < 0) || (index > 2), Exception,
-                "Expected marker scale factor index to be in the range [0, 2], "
-                "but received '{}'.", index);
+        "Contact group associated with external force '{}' not found.",
+        externalForceName);
 
     // Update the scale factor map so we can retrieve the correct MocoScaleFactor
     // for this contact group during initialization.
-    std::pair<std::string, int> key(externalForceName, index);
+    std::string key(externalForceName);
     m_scaleFactorMap[key] = name;
```

Comment:
I think you can just do this here: `m_scaleFactorMap[externalForceName] = name;`.